src/frontends/onnx/tests/onnx_import.in.cpp

```diff
@@ -6686,3 +6686,30 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_reduce_min_20_boolean) {
     test_case.add_expected_output(expected_output);
     test_case.run();
 }
+
+OPENVINO_TEST(${BACKEND_NAME}, onnx_model_multinomial_7) {
+    auto model = convert_model("multinomial.onnx");
+
+    auto test_case = ov::test::TestCase(model, s_device);
+
+    auto expected_shape = Shape{3, 5};
+    EXPECT_EQ(model->get_output_shape(0), expected_shape);
+
+    std::vector<float> input_values = {0.1f, 0.2f, 0.7f, 0.2f, 0.4f, 0.4f, 1.0f, 0.0f, 0.0f};
+    test_case.add_input<float>(ov::Shape{3, 3}, input_values);
+
+    // Values are collected for seed 1.23
+    if (std::string("${BACKEND_NAME}") == std::string("INTERPRETER")) {
+        test_case.add_expected_output<int32_t>(Shape{3, 5}, {0, 2, 1, 0, 0, 2, 1, 0, 0, 0, 0, 2, 1, 0, 0});
+    } else if (std::string("${BACKEND_NAME}") == std::string("IE_CPU")) {
+        // CPU returns a wrong values, a bug
+        // test_case.add_expected_output<int32_t>(Shape{3, 5}, {0, 2, 1, 0, 0, 2, 1, 0, 0, 0, 0, 2, 1, 0, 0});
+        GTEST_SKIP();
+    } else if (std::string("${BACKEND_NAME}") == std::string("IE_GPU")) {
```

Comment:
How were those values obtained? The test case `sample_size` (=5) is bigger than the input `class_size` dimension (=3), while the Multinomial `with_replacement` flag is set to `false`, what seems like undefined behaviour by definition.

Has the correct value for `with_replacement` flag for ONNX Multinomial case been verified? Shouldn't it be `true`?
https://github.com/openvinotoolkit/openvino/blob/0cd471032174b676c9a304309c5d61b086bf2739/docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/generation/multinomial-13.rst?plain=1#L90-L97

Reply:
Good catch. I made such test that way because I didn't find limitations on samples-size in a ONNX documentation. Even if we have wrong multinomial implementation - let's align calls by using "true" flag.